### ISSUE TITLE
untilDestroyed(this)) not working on route change

### ISSUE THREAD

**Issue by sujeetjaiswara** [ngneat/until-destroy#123]
I just trying to unsubscribe on route change but its not working

```js
@UntilDestroy({ checkProperties: true })
@Component({
   .
   .
   .
    changeDetection: ChangeDetectionStrategy.OnPush
})

ngOnInit(): void {
interval(1000)
     .pipe(untilDestroyed(this))
     .subscribe((count) => {
         console.log('count:', count);
     });
}
```

**Comment by arturovt** [comment]
Could you provide a _minimal reproducible example_?

**Comment by Smip** [comment]
I had this problem with angular 9 and @ngneat/until-destroy@8.0.1
There are no problems with @ngneat/until-destroy@7.3.2

**Comment by arturovt** [comment]
`@8.0.1` is meant to be used with Angular 10, not 9.

**Comment by Smip** [comment]
it would be nice to add this information to the readme

**Comment by arturovt** [comment]
When you install the package it says: `warning " > @ngneat/until-destroy@8.0.1" has incorrect peer dependency "@angular/core@>=10.0.5".`

**Comment by arturovt** [comment]
@sujeetjaiswara will you provide a reproducible example?

**Comment by BojanKogoj** [comment]
I had the same issue. It was as @arturovt suggested, issue with `@angular/core` version. I updated from 10.0.3 to 10.0.12 and it works.

**Comment by digeomel** [comment]
> `@8.0.1` is meant to be used with Angular 10, not 9.

@arturovt I agree with @Smip , this should be added to the readme. We have lots of warnings when doing npm install and these can be easily missed. Just lost an hour+ trying to find out what's going on.

**Comment by arturovt** [comment]
People often skip readme :)

I'm wondering if some `postinstall` hook with version check would be more obvious:
```js
const { VERSION } = require('@angular/core');
const semver = require('semver');

if (!semver.satisfies(VERSION.full, '>=10.0.5')) {
  throw new Error('@ngneat/until-destroy requires Angular@10.0.5+')
}
```

**Comment by digeomel** [comment]
@arturovt when I read the readme, I see that there are 2 versions of until-destroy, one for Ivy and one for pre-Ivy (i.e. Angular < 9).

But actually there are 3 versions. One pre-Ivy, one for Angular 9, and one for Angular 10. This is definitely not clear.

**Comment by arturovt** [comment]
@digeomel could you have a look at PR #125 ?

**Comment by arturovt** [comment]
I've published `@ngneat/until-destroy@8.0.2` with updated readme.
